test/kmeans.jl

```diff
@@ -32,6 +32,15 @@ end
     end
 end
 
+@testset "k=1 and k=n corner cases" begin
+    Random.seed!(34568)
+    @test length(kmeans(randn(2,3), 1).centers) == 2
+    @test length(kmeans(randn(2,3), 3).centers) == 6
+    x = [[ 0.5 1 ]; [ 1 0.5 ]]
+    @test dropdims(kmeans(x,1).centers, dims=2) == [0.75, 0.75]
```

Comment:
Since we have already checked (in the previous testset) that `kmeans()` returns KmeansResults, we can do it like
```suggestion
    kmeans1 = kmeans(x, 1)
    @test kmeans1.centers == reshape([0.75, 0.75], (2, 1))
```
You need also to test that the other fields are set correctly (*counts*, *assignments*, *wcounts*).
Also you need to add the weighted version test as well and check that the weights are actually used (*wcounts* should be different than the unweighted version).